src/mozillavpn.cpp

```diff
@@ -394,13 +394,23 @@ void MozillaVPN::maybeStateMain() {
   SettingsHolder* settingsHolder = SettingsHolder::instance();
 
 #if !defined(MZ_ANDROID) && !defined(MZ_IOS)
-  if (!settingsHolder->postAuthenticationShown()) {
+  if (Feature::get(Feature::Feature_newOnboarding)->isSupported()) {
+    if (!settingsHolder->onboardingCompleted()) {
+      if (!settingsHolder->onboardingStarted()) {
+        settingsHolder->setGleanEnabled(false);
+        settingsHolder->setOnboardingStarted(true);
+      }
+      setState(StateOnboarding);
+      return;
+    }
+  } else if (!settingsHolder->postAuthenticationShown()) {
     setState(StatePostAuthentication);
     return;
   }
 #endif
 
-  if (!settingsHolder->telemetryPolicyShown()) {
+  if (!settingsHolder->telemetryPolicyShown() &&
+      !Feature::get(Feature::Feature_newOnboarding)->isSupported()) {
```

Comment:
May be more readable if this feature check was moved outside the settingsHolder->telemetryPolicyShown() check:

```if (!Feature::get(Feature::Feature_newOnboarding)->isSupported())
{
    Old code.
}```